components/encoder/incremental/incremental_encoder.go

```diff
@@ -227,8 +231,6 @@ func (e *Encoder) Start(ctx context.Context) {
 	e.activeBackgroundWorkers.Add(1)
 
 	utils.ManagedGo(func() {
-		defer e.A.RemoveCallback(chanA)
-		defer e.B.RemoveCallback(chanB)
```

Comment:
Where do these lines get replaced? How does the board know to remove the callbacks when this component shuts down?

Reply:
The callbacks are now removed from the board automatically once the server stream closes. (line 232 in `board/server.go` which calls the helper function located in `board.go`. So when the component closes the `cancelCtx` passed into the `streamTicks ` gets canceled and the server will return and callbacks get removed.

Reply:
That also answers my question about closing the channel. But can you confirm a healthy viam-server shutdown and run tests from within the packages with a race flag.

Reply:
Dang, that's really smooth! Thanks for explaining, and very nice work here!

Reply:
(talked to Rand offline) Realized this will not work when using both a builtin board and builtin component since the StreamTicks call won't go through the server code. I added a call to `RemoveCallbacks` in each driver to clean up the callbacks, but we can still keep the call in server.go so that module writers don't need to worry about calling it.

Reply:
I'm going to speak a heresy. Why do we have separate implementations for built-in and modular interactions? I wonder if it would be simpler if the builtin-to-builtin pathway also went through the RPCs (with the added benefit that we'll notice more easily if the module-related pathway has some unexpected problem, because now we'd use it all the time). but I don't have a sense of how much work that would require. I trust your judgment here; maybe keep going with your current plan.